src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -713,15 +716,16 @@ internal override bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArrayS
 
             // The checks above are supposed to handle the vast majority of cases.
             // More complicated cases are handled in a special helper to make the common case scenario simple/fast
```

Comment:
I think I have added the comment. It was separation of a fast path from slow path perf tweak. It was originally a gigantic method with huge stack frame(lots of locals). Every time the method is called the stack frame must be cleaned and it was a very frequently called method. Separating fast and common part from the rest was a noticeable improvement.
I wonder if it has not been negated by all the changes that happened here since then...